### ISSUE TITLE
what does expected test failure mean?

### ISSUE THREAD

**Issue by cosmicexplorer** [jrblevin/markdown-mode#52]
Noticed this when I was running tests previously. What does `:expected-result :failed` mean? It's kinda confusing when I view the test results and see "failures: 4", even when they're expected. Does that mean the functionality hasn't been implemented yet? If it's in place if `(should-not`, I would think it would make more sense to just do that.

``` emacs-lisp
(ert-deftest test-markdown-font-lock/blockquote-link ()
  "Test font lock for links inside of a blockquote."
  :expected-result :failed
  (markdown-test-string
   "> [link](url)"
   (markdown-test-range-has-face 1 13 markdown-blockquote-face)
   (markdown-test-range-has-face 3 8 markdown-link-face)
   (markdown-test-range-has-face 9 13 markdown-url-face)))

(ert-deftest test-markdown-font-lock/blockquote-comment ()
  "Test font lock for comments inside of a blockquote."
  :expected-result :failed
  (markdown-test-string
   "> <!-- comment -->"
   (markdown-test-range-has-face 1 18 markdown-blockquote-face)
   (markdown-test-range-has-face 3 18 markdown-comment-face)))

;; and others
```

**Comment by jrblevin** [comment]
Anything with `:expected-result :failed` is known incorrect behavior. In the two cases you mentioned, the first is a reminder that I'd like to do font lock for inline links inside of blockquotes (at present, the blockquote face takes precedence). The second case actually seems to be implemented now, but I forgot to update the case (the face at position 1 should be `markdown-markup-face` rather than `markdown-blockquote-face'). I updated the description of the first one and fixed the second one in commit 1b21b1f. The two other known failing tests are related to known bugs in filling behavior. Thanks for bringing this up. Let me know if anything else isn't clear.

**Comment by cosmicexplorer** [comment]
Nope, that makes sense! Thanks for clarifying.
